eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/eventlist/itemsprovider/EventListItemsProviderImpl.java

```diff
@@ -81,18 +80,15 @@
     private Map<ScrollDirection, TaskParam> mErrorTaskParam;
 
 
-    public EventListItemsProviderImpl(FutureServiceConnection<ApiService> apiService,
-                                      EventListItems items, EventListBackgroundMessage backgroundMessage,
-                                      EventListLoadingIndicatorOverlay loadingIndicatorOverlay)
+    public EventListItemsProviderImpl(FutureServiceConnection<ApiService> apiService, EventListItems items, EventListBackgroundMessage backgroundMessage, EventListLoadingIndicatorOverlay loadingIndicatorOverlay)
     {
         mApiService = apiService;
         mBackgroundMessage = backgroundMessage;
         mItems = items;
-        mBackgroundMessage.setOnClickListener(this);
         mLoadingIndicatorOverlay = loadingIndicatorOverlay;
+        mBackgroundMessage.setOnClickListener(this);
         mExecutorService = Executors.newSingleThreadExecutor();
         mDownloadTaskClient = new DownloadTaskClient();
-        mLastResultPage = new HashMap<>(2);
         mErrorTaskParam = new HashMap<>(2);
```

Comment:
Creating  HashMap with size 2 can only hold 1 element without having to rehash. Is that really what you want?